worlds/timespinner/__init__.py

```diff
@@ -177,56 +205,57 @@ def create_item(self, name: str) -> Item:
-                and not self.is_option_enabled("UnchainedKeys"):
-            item.classification = ItemClassification.filler
+        if hasattr(self, 'options'): # self.options is not always available in this method
+            if (name == 'Tablet' or name == 'Library Keycard V') and not self.options.downloadable_items:
+                item.classification = ItemClassification.filler
+            elif name == 'Oculus Ring' and not self.options.eye_spy:
+                item.classification = ItemClassification.filler
+            elif (name == 'Kobo' or name == 'Merchant Crow') and not self.options.gyre_archives:
+                item.classification = ItemClassification.filler
+            elif name in {"Timeworn Warp Beacon", "Modern Warp Beacon", "Mysterious Warp Beacon"} \
+                    and not self.options.unchained_keys:
+                item.classification = ItemClassification.filler
 
         return item
 
     def get_filler_item_name(self) -> str:
-        trap_chance: int = self.get_option_value("TrapChance")
-        enabled_traps: List[str] = self.get_option_value("Traps")
+        trap_chance: int = self.options.trap_chance.value
+        enabled_traps: List[str] = self.options.traps.value
 
-        if self.multiworld.random.random() < (trap_chance / 100) and enabled_traps:
-            return self.multiworld.random.choice(enabled_traps)
+        if self.random.random() < (trap_chance / 100) and enabled_traps:
+            return self.random.choice(enabled_traps)
         else:
-            return self.multiworld.random.choice(filler_items) 
+            return self.random.choice(filler_items) 
 
     def get_excluded_items(self) -> Set[str]:
         excluded_items: Set[str] = set()
 
-        if self.is_option_enabled("StartWithJewelryBox"):
+        if self.options.start_with_jewelry_box:
             excluded_items.add('Jewelry Box')
-        if self.is_option_enabled("StartWithMeyef"):
+        if self.options.start_with_meyef:
             excluded_items.add('Meyef')
-        if self.is_option_enabled("QuickSeed"):
+        if self.options.quick_seed:
             excluded_items.add('Talaria Attachment')
 
-        if self.is_option_enabled("UnchainedKeys"):
+        if self.options.unchained_keys:
             excluded_items.add('Twin Pyramid Key')
 
-            if not self.is_option_enabled("EnterSandman"):
+            if not self.options.enter_sandman:
                 excluded_items.add('Mysterious Warp Beacon')
         else:
             excluded_items.add('Timeworn Warp Beacon')
             excluded_items.add('Modern Warp Beacon')
             excluded_items.add('Mysterious Warp Beacon')
 
-        for item in self.multiworld.precollected_items[self.player]:
-            if item.name not in self.item_name_groups['UseItem']:
-                excluded_items.add(item.name)
+        for item_name in self.options.start_inventory.value.keys():
```

Comment:
I think the old way was more correct here. I'm not aware of what else might put something into precollected, but if you opt in to start_inventory_from_pool this fails, and I don't trust this be safe.

Reply:
this is fine

Reply:
3rding that I also don't really understand this change

Reply:
4thing this, the change seems strange. Why was it made?

Reply:
oke ill revert this one